### ISSUE TITLE
new clients/resources do not find the resources created in Setup()

### ISSUE THREAD

**Issue by lynkfox** [spulec/moto#4831]
Versions: 
moto 3.0.2
boto3 1.20.24
pytest 6.2.5
 
 
 with the update past 3.0, any of our tests using moto failed with "Resource not Found"
 
 The tests originally looked something like this (which worked under 2.3.2:

 (create table creates the table in use and add a single item to, and returns a dynamodb resource)

 ```
 @mock_dynamodb2
class Test_from_dynamo():
    def setup(self):
        self.dynamodb = create_table()

    def teardown(self):
        # self.dynamodb.Table(TEST_TABLE_NAME).delete()
       pass

    def test_mock_data_in_dynamo(self):
        table = self.dynamodb.Table(TEST_TABLE_NAME)
        all_items = table.scan()
        assert len(all_items["Items"]) == 1
```

please note the recreation of the boto3 resource inside the test.

I am aware of the teardown workaround, however the test is failing with Resource Not Found on the table=self.dynamodb.Table(TEST_TABLE_NAME) line.

so resources set up in setup are not sticking around  into the test itself.

however, if we change the test to:

```
def test_mock_data_successfully_setup_for_comparing_premiums(self):
        dynamodb = create_table()
        table = dynamodb.Table(TEST_TABLE_NAME)
        all_items = table.scan()
        assert len(all_items["Items"]) == 1
```

The test now passes.

With over 1700 unit tests across 40 some files, most of them dealing with boto3/moto resources in some way, is there some work around that does not involve me updating every single test to call the creation driectly? Or is there some other issue here?

(of note, we've just gone back to 2.3.2 for now but would like to be able to keep our libs up to date.)

**Comment by bblommers** [comment]
Hi @lynkfox, thanks for raising this. Looks like that's a bug on our side - we'll push out a fix shortly.

This does work for classes that extend `unittest.TestCase`, where the setup-method should be called `setUp` (uppercase U), but not for plain classes where the `setup`-method should have a lowercase u.

**Comment by lynkfox** [comment]
Ah! Good to know. Updating to setUp to prevent similar issues with any other mocking libraries will be on my agenda next week. Easy fix. Thanks!

**Comment by lynkfox** [comment]
Ah! Good to know. Thanks!

**Comment by bblommers** [comment]
Hi @lynkfox, thanks again for raising this - this should now be fixed as of moto >= 3.0.3.dev8. Please let us know if you're running into any other issues!
